src/store/modules/resetpassword/actions/index.js

```diff
@@ -0,0 +1,62 @@
+import client from '@config/client';
+import { getErrorResponse } from '@utils/getResponse';
+import actionTypes from '../actionTypes';
+
+const {
+  PASSWORD_RESET_PENDING,
+  PASSWORD_RESET_SUCCESS,
+  PASSWORD_RESET_FAILED
+} = actionTypes;
+
+export const resetPasswordPending = () => {
+  return {
+    type: PASSWORD_RESET_PENDING,
```

Comment:
consider setting error to null and isSuccess to false at this point. This would help with separation of concerns and probable bugs resulting from previous requests that might already have assigned values to this object keys.

```export const resetPasswordPending = () => {
  return {
    type: PASSWORD_RESET_PENDING,
    payload: { isPending: true, error: null, isSuccess: false}
  }
}
```

Apply this change to other actions in this module appropriately where needed

Reply:
Object.assign allows you to copy the previous states and only alter what needs to change when an action is dispatched.

Reply:
Yhh, I understand that. This has nothing to do with Object.assign but the `payload`